gcc/config.gcc

```diff
@@ -1264,24 +1264,40 @@ aarch64*-*-linux*)
 	TM_MULTILIB_CONFIG=`echo $TM_MULTILIB_CONFIG | sed 's/^,//'`
 	;;
 aarch64*-*-mingw*)
-# 	tm_file="${tm_file} dbxelf.h elfos.h gnu-user.h linux.h glibc-stdint.h"
-# 	tm_file="${tm_file} aarch64/aarch64-elf.h aarch64/aarch64-errata.h aarch64/aarch64-linux.h"
-# 	tm_file="${tm_file} i386/unix.h i386/bsd.h i386/gas.h dbxcoff.h i386/cygming.h"	
-	tm_file="${tm_file} aarch64/aarch64-coff.h"
-	tm_file="${tm_file} aarch64/cygming.h"
+	tm_file="${tm_file} aarch64/aarch64-coff.h aarch64/cygming.h" # i386/unix.h i386/bsd.h i386/gas.h
+	xm_file=i386/xm-mingw32.h
+	c_target_objs="${c_target_objs} winnt-c.o"
+	cxx_target_objs="${cxx_target_objs} winnt-c.o"
+	d_target_objs="${d_target_objs} winnt-d.o"
+	target_has_targetcm="yes"
+	target_has_targetdm="yes"
+	#rust_target_objs="${rust_target_objs} winnt-rust.o"
+	#target_has_targetrustm="yes"
+	need_64bit_isa=yes
+	if test x$enable_threads = xposix ; then
+		tm_file="${tm_file} i386/mingw-pthread.h"
+	fi
+	if test x$enable_threads = xmcf ; then
+		tm_file="${tm_file} i386/mingw-mcfgthread.h"
+	fi
 	tm_file="${tm_file} aarch64/mingw32.h"
-	tm_file="${tm_file} aarch64/mingw-stdint.h"
-	tmake_file="${tmake_file} aarch64/t-aarch64" # aarch64/t-aarch64-linux"
-# 	tm_defines="${tm_defines}  TARGET_DEFAULT_ASYNC_UNWIND_TABLES=1"
 	user_headers_inc_next_post="${user_headers_inc_next_post} float.h"
+	user_headers_inc_next_pre="${user_headers_inc_next_pre} stddef.h stdarg.h"
+	tm_file="${tm_file} i386/mingw-w64.h"
+	tm_file="${tm_file} aarch64/mingw-stdint.h"
+	tmake_file="${tmake_file} t-winn aarch64/t-cygming t-slibgcc"
```

Comment:
The main part of the fix is adding `t-slibgcc` which enables `ENABLE_SHARED_LIBGCC` which enables proper linker configuration in `gcc/config/aarch64/mingw32.h`